File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private int ans = 0;
    private void traversal(TreeNode root, int max  ){
        if (root == null){
            return;
        }
        if (max<=root.val){
            ans += 1;
        }

        int newMax = Math.max(max,root.val);
        traversal(root.right,newMax);
        traversal(root.left,newMax);

    }
    public int goodNodes(TreeNode root) {
        traversal(root,Integer.MIN_VALUE);
        return ans; 
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, p: List[int]) -> int:
        min_price = p[0]
        max_profit = 0

        for i in range(1, len(p)):
            if p[i] < min_price:
                min_price = p[i]
            else:
                max_profit = max(max_profit, p[i] - min_price)

        return max_profit
        